### ISSUE TITLE
msvc14.0 x64 build - Client crash on Channel::Create()

### ISSUE THREAD

**Issue by kuroburki** [alanxz/SimpleAmqpClient#131]
I built the SimpleAmqpClient for msvc14.0 x64 configuration as a shared library. The very first invocation of the library routine (Channel::Create()) throws a runtime exception and aborts. Anyone else facing similar problem? The same code builds and works fine as expected on RHEL with gcc toolset. Looks to me like a build configuration issue.

**Comment by kuroburki** [comment]
I had to modify SimpleAmqpClient/Channel.h and Channel.cpp files to fix this issue. Function details for Channel::Create() had to be moved into the source file from header. I verified this issue shows up with msvc12.0 build as well. There is an additional unresolved symbol issue while using the string constants from Channel - EXCHANGE_TYPE_DIRECT, EXCHANGE_TYPE_TOPIC, EXCHANGE_TYPE_FANOUT will not be resolved when used in my client code. This has something to do with declspec(dllimport), but I have not been able to figure out as of this morning.
